google-cloud-bigtable/lib/google/cloud/bigtable/config.rb

```diff
@@ -0,0 +1,72 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Google
+  module Cloud
+    module Bigtable
+      class Config
```

Comment:
I don't understand the role this `Config` class plays. Is it part of the public API? The `project_id` value is already set on `InstanceAdminClient`, `TableAdminClient`, and `DataClient` in the `Google::Cloud::Bigtable.new` method. How does it get passed to `Config`?

Reply:
Config class is mostly a wrapper of common setup functions and config. The class is created so that i do not have to create large initializers in each set classes. 

eg. the class is passed to InstanceSet class as we call bigtable.instances which internally creates and maintains an InstanceAdmin client. Similar usage for TableSet class.